packages/payment/src/services/payment-module.ts

```diff
@@ -409,33 +416,55 @@ export default class PaymentModuleService<
   ): Promise<PaymentDTO> {
     const payment = await this.paymentService_.retrieve(
```

Comment:
Maybe we don't need to handle it right away, but esp. when it comes to payments we need to ensure there are no race conditions (and currently there are few scenarios where they can occur). Just something to keep in mind